pyproject.toml

```diff
@@ -11,6 +11,7 @@ ipykernel = "^6.29.5"
 scikit-learn = "^1.5.2"
 pandas = "^2.2.3"
 numpy = "^2.1.2"
+pre-commit = "^4.0.1"
```

Comment:
pre-commit should be installed globally
your system should already have it when I setup your laptop